go-controller/pkg/cluster/master.go

```diff
@@ -344,18 +353,25 @@ func (cluster *OvnClusterController) addNode(node *kapi.Node) error {
 			return nil
 		}
 	}
-	// Create new subnet
-	sn, err := cluster.masterSubnetAllocator.GetNetwork()
-	if err != nil {
-		return fmt.Errorf("Error allocating network for node %s: %v", node.Name, err)
-	}
 
-	err = cluster.Kube.SetAnnotationOnNode(node, OvnHostSubnet, sn.String())
-	if err != nil {
-		_ = cluster.masterSubnetAllocator.ReleaseNetwork(sn)
-		return fmt.Errorf("Error creating subnet %s for node %s: %v", sn.String(), node.Name, err)
+	// Create new subnet
+	for _, possibleSubnet := range cluster.masterSubnetAllocatorList {
+		sn, err := possibleSubnet.GetNetwork()
+		if err == netutils.ErrSubnetAllocatorFull {
+			// Current subnet exhausted, check next possible subnet
+			continue
+		} else if err != nil {
+			return fmt.Errorf("Error allocating network for node %s: %v", node.Name, err)
+		} else {
+			err = cluster.Kube.SetAnnotationOnNode(node, OvnHostSubnet, sn.String())
+			if err != nil {
+				_ = possibleSubnet.ReleaseNetwork(sn)
+				return fmt.Errorf("Error creating subnet %s for node %s: %v", sn.String(), node.Name, err)
+			}
+			logrus.Infof("Created HostSubnet %s", sn.String())
+			return nil
+		}
 	}
-	logrus.Infof("Created HostSubnet %s", sn.String())
 	return nil
```

Comment:
@JacobTanenbaum shouldn't we return an error here saying that all subnets have been exhausted?